auth/middleware.go

```diff
@@ -102,7 +102,10 @@ func (k *kratosMiddleware) Session(next echo.HandlerFunc) echo.HandlerFunc {
 				email = e
 			}
 		}
-		ctx.WithUser(&api.ContextUser{ID: uuid.MustParse(session.Identity.GetId()), Email: email})
+
+		if uid, err := uuid.Parse(session.Identity.GetId()); nil == err {
```

Comment:
nit: `err == nil`

Reply:
If parsing fails, 4xx error unauthorized should be returned, auth being passed means that a user-id is expected in further functions

Reply:
> `nit: err == nil`

This is actually deliberate. `err != nil` is so prevalent in Go that even `err == nil` might appear as `err != nil` on a glance. `nil == err` makes it noticeable in my opinion.

> If parsing fails, 4xx error unauthorized should be returned, auth being passed means that a user-id is expected in further functions

Had to skip auth on `/kratos/*` calls other wise if we return 401, the login page would break. I am not entirely sure if this is okay.

![image](https://github.com/flanksource/incident-commander/assets/13419481/0c5bee54-74b3-481e-b8e1-51d28c90ba7d)

Reply:
Can you confirm with @mainawycliffe once how frontend should behave in these scenarios

Reply:
@yashmehrotra @adityathebe There are a few endpoints I would expect to be public - login, registration (not currently in use), password reset, etc., but others such as user profile related would probably need to be authenticated.

Reply:
@yashmehrotra you think it's fine to skip auth on all `/kratos/*` calls?

Reply:
> This is actually deliberate. err != nil is so prevalent in Go that even err == nil might appear as err != nil on a glance. nil == err makes it noticeable in my opinion.

I don't see why nil == err is any different from err == nil,  Go is meant to be un-opionated so that when you read code you don't need to think about formatting and nil == err, is non-standard and makes you think so even if it is technically better in isolation, its deviation from the standard makes it worse